lib/internal/modules/esm/hooks.js

```diff
@@ -497,13 +438,115 @@ class Hooks {
       __proto__: null,
       format,
       responseURL,
-      source,
+      source: ''+source,
     };
   }
 }
-
 ObjectSetPrototypeOf(Hooks.prototype, null);
 
+class HooksProxy {
+  /**
+   * The request & response segment of the shared memory. TextEncoder/Decoder (needed to convert
+   * requests & responses into a format supported by the comms channel) reads and writes with
+   * Uint8Array.
+   */
+  #data;
+  /**
+   * The lock/unlock segment of the shared memory. Atomics require this to be a Int32Array. This
+   * segment is used to tell the main to sleep when the worker is processing, and vice versa
+   * (for the worker to sleep whilst the main thread is processing).
+   * 0 -> main sleeps
+   * 1 -> worker sleeps
+   */
+  #lock;
+
+  #isReady = false;
+
+  /**
+   * @param {import('./loader.js).KeyedExports} customLoaders Exports from user-defined loaders
+   *  (as returned by `ModuleLoader.import()`).
+   */
+  constructor() {
+    const { InternalWorker } = require('internal/worker');
+
+    const data = new SharedArrayBuffer(204_800);
+    const lock = new SharedArrayBuffer(4);
+    this.#data = new Uint8Array(data);
+    this.#lock = new Int32Array(lock);
+
+    const worker = this.worker = new InternalWorker('internal/modules/esm/worker', {
+      stderr: false,
+      stdin: false,
+      stdout: false,
+      trackUnmanagedFds: false,
+      workerData: {
+        data,
+        lock,
+      },
+    });
+    worker.unref(); // ! Allows the process to eventually exit when worker is in its final sleep.
+  }
+
+  makeRequest(method, ...args) {
+    if (!this.#isReady) {
+      const { kIsOnline } = require('internal/worker');
+      if (!this.worker[kIsOnline]) {
+        Atomics.wait(this.#lock, 0, 0); // ! Block the main thread until the worker is ready.
+      }
+
+      this.#isReady = true;
+    }
+
+    TypedArrayPrototypeFill(this.#data, undefined); // Erase handled request/response data
+// console.log('sending request', method, args)
+    const request = serialize({ method, args });
+    TypedArrayPrototypeSet(this.#data, request);
+
+    // this.#awaitResponse();
+
+    const chunks = [];
+    let done = false;
+    try {
+      while (done === false) {
+        this.#awaitResponse();
+
+        const chunk = deserialize(this.#data);
+        console.log('chunk:', chunk)
+
+        if (chunk instanceof Error) { throw chunk; }
+
+        ({ done } = chunk);
+        chunks.push(chunk.value);
+
+        console.log('awaiting next')
+      }
+    } catch (exception) {
+      console.log('\nmakeRequest failed!\n');
+      console.log(this.#data);
+      console.log('byteLength:', this.#data.byteLength);
+      console.log('this.#data[0]:', this.#data[0]);
+      if (this.#data[0] === 0) { // Response should not be empty
```

Comment:
`data` is set starting at index `0`, so even if something else went wrong, we should only need to check the first index.